[AOTI] Delay the fallback kernel naming decision to the codegen time

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113762
* __->__ #113660

Summary: This is to prepare for a later change that changes AOTI's second-pass to perform codegen only.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D51382677](https://our.internmc.facebook.com/intern/diff/D51382677)

desertfire committed:
[AOTI] Delay the fallback kernel naming decision to the codegen time

Summary: This is to prepare for a later change that changes AOTI's second-pass to perform codegen only.

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Delay the fallback kernel naming decision to the codegen time"


Summary: This is to prepare for a later change that changes AOTI's second-pass to perform codegen only.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Delay the fallback kernel naming decision to the codegen time"


Summary: This is to prepare for a later change that changes AOTI's second-pass to perform codegen only.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Delay the fallback kernel naming decision to the codegen time"


Summary: This is to prepare for a later change that changes AOTI's second-pass to perform codegen only.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -4015,10 +4028,10 @@ def codegen(self, wrapper):
             x,
             [x, self.constant_args[0], index, src],
-            self.kernel,
+            self.cpp_kernel if V.graph.cpp_wrapper else self.kernel,
```
Can we call ```self.codegen_kernel_name``` here?

desertfire replied:
This is because `codegen_kernel_name` doesn't exist in the base class `ExternKernel`, since not all `ExternKernel` has a kernel name. I think @int3 raised a similar question in a different context, `codegen_kwargs`. I will create some other PRs to do some refactoring to the base/child classes here.

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -4110,7 +4115,11 @@ def codegen(self, wrapper):
-        wrapper.writeline(wrapper.wrap_kernel_call(self.kernel, args))
+        wrapper.writeline(
+            wrapper.wrap_kernel_call(
+                self.cpp_kernel if V.graph.cpp_wrapper else self.kernel, args
```
ditto - can we call ```self.codegen_kernel_name```?

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -4925,7 +4936,7 @@ def __init__(
         wrapper.generate_extern_kernel_alloc_and_find_schema_if_needed(
             self.get_name(),
-            self.kernel,
+            self.cpp_kernel if V.graph.cpp_wrapper else self.kernel,
```
ditto

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -3718,7 +3718,7 @@ def codegen(self, wrapper):
             self.codegen_reference(),
             args,
-            self.kernel,
+            self.cpp_kernel if V.graph.cpp_wrapper else self.kernel,
```
same

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51382677).

chenyang78 reviewed (approved):
LGTM. Thanks!

desertfire committed:
Update on "[AOTI] Delay the fallback kernel naming decision to the codegen time"


Summary: This is to prepare for a later change that changes AOTI's second-pass to perform codegen only.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

Differential Revision: [D51382677](https://our.internmc.facebook.com/intern/diff/D51382677)

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI] Delay the fallback kernel naming decision to the codegen time"


Summary: This is to prepare for a later change that changes AOTI's second-pass to perform codegen only.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

Differential Revision: [D51382677](https://our.internmc.facebook.com/intern/diff/D51382677)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51382677).
